standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -675,7 +867,24 @@ public boolean openTransaction() {
 
   @Override
   public boolean commitTransaction() {
-    return rawStore.commitTransaction();
+    if (!rawStore.commitTransaction()) {
+      return false;
+    }
+
+    // In case of event based update, shared cache is not updated directly to avoid inconsistency.
+    // For example, if metastore B add a partition, then metastore A drop a partition later. However, on metastore A,
+    // it first get drop partition request, then from notification, create the partition. If there's no tombstone
+    // entry in partition cache to tell drop is after creation, we end up consumes the creation request. Though
+    // eventually there's drop partition notification, but during the interim, later event takes precedence.
+    // So we will not update the cache during raw store operation but wait during commit transaction to make sure that
+    // the event related to the current transactions are updated in the cache and thus we can support strong
+    // consistency in case there is only one metastore.
+    // TODO : There will be an issue if the cache is accessed to get the object updated within the same transaction.
```

Comment:
I see you will get from ObjectStore when this happen. Is the TODO still applicable?

Reply:
yes, the todo can be removed